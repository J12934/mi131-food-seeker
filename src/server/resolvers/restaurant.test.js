const { Query: { search, typeahead, restaurant } } = require('./restaurant');

jest.mock('../services/RestaurantAPI');

describe('restaurant resolvers', () => {
    describe('search()', () => {
        it('retuns expected result', async () => {
            const result = await search(null, {
                term: 'burger',
                coordinates: {
                    latitude: 54,
                    longitude: 11,
                },
            });

            expect(result).toMatchSnapshot();
        });
    });

    describe('typeahead()', () => {
        it('retuns expected result', async () => {
            const result = await typeahead(null, {
                term: 'bur',
                coordinates: {
                    latitude: 54,
                    longitude: 11,
                },
            });

            expect(result).toMatchSnapshot();
        });
    });

    describe('restaurant()', () => {
        it('retuns expected result', async () => {
            const result = await restaurant(null, { id: 'johns-burgers-kiel' });

            expect(result).toMatchSnapshot();
        });
    });
});
