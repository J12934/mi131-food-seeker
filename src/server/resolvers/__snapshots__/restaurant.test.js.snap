// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`restaurant resolvers restaurant() retuns expected result 1`] = `
Object {
  "coordinates": Object {
    "latitude": 54.33204,
    "longitude": 10.12449,
  },
  "id": "johns-burgers-kiel",
  "name": "John's Burgers",
  "rating": 4.5,
}
`;

exports[`restaurant resolvers search() retuns expected result 1`] = `
Array [
  Object {
    "coordinates": Object {
      "latitude": 54.33204,
      "longitude": 10.12449,
    },
    "id": "johns-burgers-kiel",
    "name": "John's Burgers",
    "rating": 4.5,
  },
  Object {
    "coordinates": Object {
      "latitude": 54.33254,
      "longitude": 10.11822,
    },
    "id": "kitty-rock-belly-full-kiel",
    "name": "Kitty Rock Belly Full",
    "rating": 4.5,
  },
]
`;

exports[`restaurant resolvers typeahead() retuns expected result 1`] = `
Array [
  "Burgers",
  "Burmese",
  "Fast Food",
]
`;
